convert2rhel/checks.py

```diff
@@ -342,7 +342,7 @@ def check_custom_repos_are_valid():
         return
 
     output, ret_code = call_yum_cmd(
-        command="makecache",
+        command="repolist",
```

Comment:
Also, before calling yum makecache, we need to be sure the cache does not exist otherwise yum would skip making it: https://github.com/oamg/convert2rhel/pull/196#discussion_r609974562.

You could tell me _"but hey, we're cleaning the cache in the `clean_yum_metadata()`"_. That's true but it cleans the cache just for the repos that are *enabled* on the system.

Now consider this scenario:
- I'm on a CentOS Linux 7 system and I'm preparing it for conversion to RHEL 7
- I have a mirror of RHEL 7 repo on an FTP server available through my internal company network
- I create a /etc/yum.repos.d/rhel7-on-our-ftp.repo file where I define a `rhel-7-server-rpms` repoid pointing to the RHEL 7 repo on my on-premise FTP server
- I make the repo disabled by default (`enabled=0`)
- I check whether the repo definition is ok by running `yum info kernel --setopt=reposdir=/home/admin/conversion/ --enablerepo=rhel-7-server-rpms` and it works ok. That command generates a yum cache for this repo
- Couple of hours later I run convert2rhel, not knowing the FTP server is down
- convert2rhel runs `yum clean metadata` but that cleans cache of just the enabled repos on the system and the `rhel-7-server-rpms` is not enabled
- convert2rhel runs this custom repo availability check in which it runs `yum makecache -v --setopt=*.skip_if_unavailable=False --disablerepo="*" --enablerepo=rhel-7-server-rpms` and it returns 0 because it uses the cache generated hours ago

@abadger, do you think is my scenario valid? If it is I'd recommend cleaning the cache either in the `clean_yum_metadata() and using `--enablerepo="*"` or by running another `yum clean metadata` right before this `yum makecache` with using the `--enablerepo` option.

Reply:
I found the suggestion in a gist from the Jira ticket. It would replace the verification of repos that makecache previously did. Is there a better command that can be run it it's place @bocekm ?

Reply:
Ah, ok. @r0x0d, why have you decided to replace makecache with repolist in https://gist.github.com/r0x0d/f645a1f7b005ee96e9421954e27528db?

Reply:
Not sure, too, @bocekm. It's been quite a while since I did this gist, but I think that my thinking back then was: 

1. We need to check for repo availability, so, calling repolist could work 
2. If we need to pass the `--enablerepo` and `--disablerepo` to that call, we could extend it to check only that. 
3. Back then, I thought that makecache would not verify the repositories, only generate the caches
4.  I slightly remember we were discussing this over some meeting and realizing that `repolist` might not be the best thing here, probably another command could do the trick for us
5.  I haven't investigated further since then

Reply:
Based on https://github.com/oamg/convert2rhel/pull/196#discussion_r588298250 it seems that `repolist` returns 0 even when there's an inaccessible repo. So I'd stick to makecache which we know returns 1 in such a case.
The `--enablerepo` is passed to the `yum makecache` within the `call_yum_cmd()` function.

Reply:
@bocekm so since we are removing the make cache call from check_custom_repos_are_valid() in checks.py, can we just get rid of that function all together?

Reply:
@bocekm and do we no longer want to generate the yum cache after cleaning the yum cache and before the system checks?

Reply:
I believe that we don't need to generate the yum cache "manually" because whatever yum/repoquery/yumdownloader command we call later triggers the cache generation automatically.

Reply:
Noted thank you